neuraxle/steps/flow.py

```diff
@@ -326,29 +287,56 @@ class ChooseOneOrManyStepsOf(FeatureUnion):
         :class:`Optional`
     """
 
-    def __init__(self, steps, hyperparams=None):
+    def __init__(self, steps):
         FeatureUnion.__init__(self, steps)
-
-        if hyperparams is None:
-            self.set_hyperparams(HyperparameterSamples({}))
-        else:
-            self.set_hyperparams(hyperparams)
-
+        self.set_hyperparams(HyperparameterSamples({}))
         self._make_all_steps_optional()
 
     def _make_all_steps_optional(self):
         """
         Wrap all steps with :class:`Optional` wrapper.
         """
         step_names = list(self.keys())
-        for step_name in step_names:
+        for step_name in step_names[:-1]:
             self[step_name] = Optional(self[step_name])
         self._refresh_steps()
 
 
 CHOICE_HYPERPARAM = 'choice'
 
 
+class SelectNotEmptyJoiner(TransformHandlerMixin, BaseStep):
```

Comment:
not sure what this is. name better or document

Reply:
SelectNonEmptyDataInputs, I don't know how to make it clearer